### PR TITLE
Remove enum34 from vendor directory

### PR DESCRIPTION
**Pull request by anshul-patel-infostretch**
- Added enum34 for < python3.4 in setup.py, Pipfile.
- Updated _compat.py to incorporate the vendor change.

## Tests

- Ran `make unit` and `make integration`, it passed.

## Note

Changes regarding to new `pylint` version has been incorporated in #27. So didn't duplicate those changes in this PR.

### CODE REVIEW

File: crux/_compat.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
     from builtins import str as unicode
-    from enum import Enum
     from urllib.parse import quote as urllib_quote  # type: ignore
 except ImportError:
     # Python 2 imports
     from __builtin__ import unicode  # type: ignore
-    from crux._vendor.enum import Enum  # type: ignore
     from urllib import quote as urllib_quote
 
+from enum import Enum
```
Comment (antoncohen) [inline crux/_compat.py]: Based on this being outside the try/except, does this mean we don't need `Enum` in compat, and modules can just `from enum import Enum` themselves?

File: Pipfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ name = "pypi"
 [packages]
 google-resumable-media = { version = "*", extras = ["requests"] }
 typing = {version = "*", markers="python_version < '3.5'"}
+enum34 = {version = "*", markers="python_version < '3.4'"}
```
Comment (antoncohen) [inline Pipfile]: What do you think about keeping these in alphabetical order (and setup.py requirements too)?

Comment (anshul-patel-infostretch) [reply Pipfile]: My bad, thanks and agreed updated the code.

File: crux/_compat.py (hunk 1 above)
Comment (anshul-patel-infostretch) [reply crux/_compat.py]: Thanks and Agreed, updated the code.

**Pull request merged**